stock_valuation_mrp/views/product_template_views.xml

```diff
@@ -0,0 +1,32 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <record id="product_template_form_view_stock_valuation_mrp" model="ir.ui.view">
+        <field name="name">product.template.form.view (in stock_valuation_mrp)</field>
+        <field name="model">product.template</field>
+        <field name="inherit_id" ref="product.product_template_only_form_view"/>
+        <field name="arch" type="xml">
+            <field name="barcode" position="after">
+                <field 
+                    name="has_waste_cost_mgmt"
+                    widget="boolean_toggle"
+                    attrs="{'invisible': [
+                        ('warehouse_valuation','=',False),
+                    ]}"
+                />
+                <field
+                    name="residue_pricelist_mgmt"
+                    attrs="{
+                        'invisible': 
+                        [
+                            '|',
+                                ('has_waste_cost_mgmt','=',False),
+                                ('warehouse_valuation','=',False)
+                        ],
```

Comment:
Why this condition is different than product variant form one? `has_waste_cost_mgmt` checking should be enough, if I'm right...

Reply:
Changing the warehouse_valuation from a product category might cause has_waste_cost_mgmt to become True even if warehouse_valuation itself is False.
Resulting in a view with has_waste_cost_mgmt as invisible, and pricelist as visible